plugins/s3_auth/s3_auth.cc

```diff
@@ -93,6 +96,22 @@ class S3Config
     return _secret && (_secret_len > 0) && _keyid && (_keyid_len > 0) && (2 == _version);
   }
 
+  void
+  acquire()
+  {
+    ink_atomic_increment(&_ref_count, 1);
+  }
+
+  void
+  release()
+  {
+    TSDebug(PLUGIN_NAME, "ref_count is %d", _ref_count);
+    if (1 >= ink_atomic_decrement(&_ref_count, 1)) {
```

Comment:
I copied / pasted this from other plugins :). I don't mind changing it though, but then we probably should change it in all ~5 of them?